### PR TITLE
morphology

### PR DESCRIPTION
**Pull request by Juclique**
### Description
A few sentences describing the changes proposed in this pull request.

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

File: docs/.gitignore [hunk 1]
```diff
@@ -1,5 +1,3 @@
-build/doctest
```
Comment (edgarriba) [inline docs/.gitignore:1]: Don't remove it

File: kornia/__init__.py [hunk 2]
```diff
@@ -112,3 +113,12 @@
 from kornia.constants import *
 
 from kornia.testing import xla_is_available
+
+from kornia.morphology import (
```
Comment (edgarriba) [inline kornia/__init__.py:117]: Let's skip this by the moment. We can expose later to too level of needed

File: test/morphology/test_black_hat.py [hunk 3]
```diff
@@ -65,14 +65,3 @@ def test_jit(self, device, dtype):
         expected = op(input, kernel)
 
         assert_allclose(actual, expected)
-
-    @pytest.mark.nn
-    def test_module(self, device, dtype):
```
Comment (edgarriba) [inline test/morphology/test_black_hat.py]: I wouldn't remove this tests

**Review by edgarriba — APPROVED** [review]

**Review by shijianjian — COMMENTED** [review]
Just one concern about the gradient graph.

File: kornia/morphology/basic_operators.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
+# Libraries
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from typing import Tuple, Union
+
+
+# _se_to_mask
+def _se_to_mask(se: torch.Tensor) -> torch.Tensor:
```
[the remark this answers is not in the document]
Comment (edgarriba) [reply kornia/morphology/basic_operators.py:9]: It won't break, just a constant I believe

**Comment by edgarriba** [comment]
@Juclique thanks for this great contribution!!! 

Could you open whenever you have time a quick PR fixing the rendering issues with docs ?
https://kornia.readthedocs.io/en/latest/morphology.html
